messenger/messenger.go

```diff
@@ -266,12 +274,15 @@ func (m *MesosMessenger) Start() error {
 
 // Stop stops the messenger and clean up all the goroutines.
 func (m *MesosMessenger) Stop() error {
+	defer m.stopOnce.Do(func() { close(m.stop) })
```

Comment:
instead of stopOnce use an atomic/int32 latch to guard execution of this entire func